Classes/Synchronization/Synchronizer.php

```diff
@@ -182,17 +188,37 @@ protected function syncUpdate(SourceFile $sourceFile, FileState $fileState)
      */
     protected function addTags(Asset $asset)
     {
+        $tags = new ArrayCollection();
         foreach ($this->source->getAssetTags() as $tagLabel) {
-            $tag = $this->tagRepository->findOneByLabel($tagLabel);
-
-            if ($tag === null) {
-                $tag = new Tag($tagLabel);
-                $this->tagRepository->add($tag);
+            if (trim($tagLabel) !== '') {
+                $tags->add($this->getOrCreateTag($tagLabel));
             }
+        }
+        $asset->setTags($tags);
```

Comment:
Wouldn't this simply overwrite the assigned tags of the asset? Is this really the intended behavior or shouldn't the new tags be better *added* to the already assigned ones (except for the case the tag is already assigned)?